src/parser/src/Type.cpp

```diff
@@ -59,3 +59,7 @@ llvm::Type* Typing::MatrixType::getLLVMType(Utils::IRContext* context) const {
     }
     return ty;
 }
+const std::vector<uint>& Typing::MatrixType::getDimensions() const { return this->dimensions; }
+
+// TODO make this not a noop
+bool Typing::MatrixType::simpleDimensionCompatible(const Typing::MatrixType& val) const { return true; }
```

Comment:
I was keeping these in the header to make it more obvious that they aren't functional yet...